modules/distributor/distributor.go

```diff
@@ -376,6 +378,12 @@ func (d *Distributor) sendToIngestersViaBytes(ctx context.Context, userID string
 		op = ring.Write
 	}
 
+	batchResults := make([][]int, len(keys))
```

Comment:
this is tough. for each trace we need to know 2 things to accurately count refused spans:

- number of successes
- first (or last or any) reported error

it's possible there is an optimization to be made here to only track these fields instead of every returned value for each trace. we'd need to do some benchmarking to determine if there was a better path. possibly something to keep in mind while we finish up.

Reply:
the only optimization I can think of which I don't know if it will be an optimization is:

```go

// instead of batchResults
NumSuccessByTraceIndex   = make([]int, len(keys)
LastPushErrorByTraceIndex = make([]int, len(keys)
```

And every time we get a PushResponse, we update each trace accordingly. What do you think?